src/Hive/Services/Common/ModService.cs

```diff
@@ -170,6 +170,46 @@ public async Task<HiveObjectQuery<Mod>> GetMod(User? user, string id, VersionRan
             return new HiveObjectQuery<Mod>(StatusCodes.Status200OK, mod);
         }
 
+        /// <summary>
+        /// Gets a <see cref="Mod"/> that matches the provided <paramref name="identifier"/>.
+        /// </summary>
+        /// <param name="user">The user associated with this request.</param>
+        /// <param name="identifier">Identifier which represents the mod.</param>
+        /// <returns>A warapped <see cref="Mod"/> of the found mod, if successful.</returns>
+        public async Task<HiveObjectQuery<Mod>> GetMod(User? user, ModIdentifier identifier)
+        {
+            if (identifier is null)
+                return new(StatusCodes.Status400BadRequest, "Mod Identifier is invalid");
+
+            // Combine plugins
+            log.Debug("Combining plugins...");
+            var combined = plugin.Instance;
+
+            if (!Version.TryParse(identifier.Version, out var modVersion))
+            {
+                return new(StatusCodes.Status400BadRequest, "Could not parse Version string");
```

Comment:
It might be worth returning an error message describing the version parsing failure.